pkg/mimir/mimir.go

```diff
@@ -756,7 +757,11 @@ func New(cfg Config, reg prometheus.Registerer) (*Mimir, error) {
 	}
 
 	mimir.setupObjstoreTracing()
-	otel.SetTracerProvider(NewOpenTelemetryProviderBridge(opentracing.GlobalTracer()))
+
+	// Injects span profiler into the tracer for cross-referencing between traces and profiles.
+	// Note, for performance reasons, span profiler only labels root spans.
+	tracer := spanprofiler.NewTracer(opentracing.GlobalTracer())
```

Comment:
I think here we are missing assigning global tracer with the wrapped tracer again opentracing.SetGlobalTracer(wrappedTracer) so that opentracing.StartSpan is going to use the wrapped spanprofiler tracer instead.

Reply:
You are right. I didn't release we're using both opentracing and opentelemetry, thank you 🆙